pandas/io/excel/_openpyxl.py

```diff
@@ -509,24 +509,14 @@ def get_sheet_by_index(self, index: int):
 
     def _convert_cell(self, cell, convert_float: bool) -> Scalar:
 
-        from openpyxl.cell.cell import TYPE_BOOL, TYPE_ERROR, TYPE_NUMERIC
+        from openpyxl.cell.cell import TYPE_ERROR, TYPE_NUMERIC
 
         if cell.value is None:
             return ""  # compat with xlrd
-        elif cell.is_date:
-            return cell.value
         elif cell.data_type == TYPE_ERROR:
             return np.nan
-        elif cell.data_type == TYPE_BOOL:
-            return bool(cell.value)
```

Comment:
#5394 claims that these values will come through as floats, has that changed? Are there tests for excel sheets with bool/int/etc?